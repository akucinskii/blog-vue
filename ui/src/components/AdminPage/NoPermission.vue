<script setup lang="ts">
import { useRouter } from "vue-router";

const router = useRouter();
</script>

<template>
  <v-main>
    <v-container>
      <v-row align="center" justify="center" class="h-100" no-gutters>
        <v-col>
          <v-spacer />
          <v-row justify="center" align="center">
            <h1 class="text-center py-8">You need to be admin to see this page</h1>
          </v-row>
          <v-row justify="center" align="center">
            <v-btn id="returnButton" @click="router.push('/')">Return to home page</v-btn>
          </v-row>
          <v-spacer />
        </v-col>
      </v-row>
    </v-container>
  </v-main>
</template>
