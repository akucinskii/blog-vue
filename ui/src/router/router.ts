import { createRouter, createWebHistory } from "vue-router";
import NavbarComponent from "../components/navbar/NavbarComponent.vue";
import CategoryView from "../views/CategoryView.vue";
import HomeView from "../views/HomeView.vue";
import PostView from "../views/PostView.vue";
import LoginView from "../views/LoginView.vue";
import RegisterView from "../views/RegisterView.vue";
import ProfileView from "../views/ProfileView.vue";
import AdminPanelView from "../views/AdminPanelView.vue";

export const routes = [
  {
    path: "/",
    name: "home",
    components: {
      default: HomeView,
      Navbar: NavbarComponent,
    },
  },
  {
    path: "/post/:id",
    name: "post",
    components: {
      default: PostView,
      Navbar: NavbarComponent,
    },
  },

  {
    path: "/category/:id",
    name: "category",
    components: {
      default: CategoryView,
      Navbar: NavbarComponent,
    },
  },

  {
    path: "/login",
    name: "login",
    components: {
      default: LoginView,
      Navbar: NavbarComponent,
    },
  },
  {
    path: "/register",
    name: "register",
    components: {
      default: RegisterView,
      Navbar: NavbarComponent,
    },
  },
  {
    path: "/profile",
    name: "profile",
    components: {
      default: ProfileView,
      Navbar: NavbarComponent,
    },
  },
  {
    path: "/admin",
    name: "admin",
    components: {
      default: AdminPanelView,
      Navbar: NavbarComponent,
    },
  },
];

const router = createRouter({
  history: createWebHistory(),
  routes,
});

export default router;
