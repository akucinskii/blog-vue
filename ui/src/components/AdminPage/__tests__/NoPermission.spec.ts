import { shallowMount, VueWrapper } from "@vue/test-utils";
import { nextTick } from "vue";
import { createRouter, createWebHistory, Router } from "vue-router";
import { routes } from "../../../router/router";
import NoPermission from "../NoPermission.vue";

let wrapper: VueWrapper<InstanceType<typeof NoPermission>>;
let router: Router;

describe("NoPermission", () => {
  beforeEach(async () => {
    router = createRouter({
      history: createWebHistory(),
      routes: routes,
    });

    router.push("/");
    await router.isReady();

    wrapper = shallowMount(NoPermission, {
      global: {
        plugins: [router],
      },
    });
  });

  test("matches snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("On click redirects to login", async () => {
    const push = jest.spyOn(router, "push");
    const returnButton = wrapper.find("#returnButton");

    await returnButton.trigger("click");
    await nextTick();

    expect(push).toHaveBeenCalledWith("/");
  });
});
