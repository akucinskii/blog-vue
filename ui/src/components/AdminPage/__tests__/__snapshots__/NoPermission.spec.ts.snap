// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPermission matches snapshot 1`] = `
<v-main>
  <v-container>
    <v-row
      align="center"
      class="h-100"
      justify="center"
      no-gutters=""
    >
      <v-col>
        <v-spacer />
        <v-row
          align="center"
          justify="center"
        >
          <h1
            class="text-center py-8"
          >
            You need to be admin to see this page
          </h1>
        </v-row>
        <v-row
          align="center"
          justify="center"
        >
          <v-btn
            id="returnButton"
          >
            Return to home page
          </v-btn>
        </v-row>
        <v-spacer />
      </v-col>
    </v-row>
  </v-container>
</v-main>
`;
